cpp/src/decisiontree/decisiontree.cu

```diff
@@ -101,9 +104,9 @@ void validity_check(const DecisionTreeParams params) {
            (params.split_algo < SPLIT_ALGO::SPLIT_ALGO_END),
          "split_algo value %d outside permitted [0, %d) range",
          params.split_algo, SPLIT_ALGO::SPLIT_ALGO_END);
-  ASSERT((params.min_rows_per_node >= 2),
+  ASSERT((params.min_samples_leaf >= 2),
```

Comment:
similar to this, do wee need a check on `min_samples_split` too?

Reply:
Will add a check for `min_samples_split`